tests/integration/validations/factory-general-test.js

```diff
@@ -247,3 +247,35 @@ test("custom messages object", function(assert) {
   assert.equal(object.get('validations.attrs.firstName.isValidating'), false);
   assert.equal(object.get('validations.attrs.firstName.message'), 'Test error message');
 });
+
+test("debounced validations", function(assert) {
+  var done = assert.async();
+  var Validations = buildValidations({
+    firstName: validator(Validators.presence),
+    lastName: validator(Validators.presence, { debounce: 500 }),
+  });
+  var object = setupObject(this, Ember.Object.extend(Validations));
+
+  assert.equal(object.get('validations.isValid'), false, 'isValid was expected to be FALSE');
+  // TODO: I feel like initially a debounced validation should not be debounced.
+  assert.equal(object.get('validations.isValidating'), true, 'isValidating was expected to be TRUE');
+  assert.equal(object.get('validations.isTruelyValid'), false, 'isTruelyValid was expected to be FALSE');
+
+  Ember.run.later(() => {
+    assert.equal(object.get('validations.attrs.lastName.isValid'), false);
+    assert.equal(object.get('validations.attrs.lastName.isValidating'), false);
+    assert.equal(object.get('validations.attrs.lastName.message'), 'lastName should be present');
+
+    object.set('lastName', 'Golan');
+    // TODO: This is true because a new Validation result object is created which initially defaults to TRUE
+    assert.equal(object.get('validations.attrs.lastName.isValid'), true);
```

Comment:
Again, this doesnt feel right. When we set a new value to the debounced CP, it creates a new ValidationResult object with the new promise but that new object gets instantiated by default with `isValid: true`